modules/gmake/gmake.lua

```diff
@@ -0,0 +1,338 @@
+				end
+				return flags
+			end,
+			getCppFlags = function (cfg)
+				local flags = {}
+
+				return flags
+			end,
+			getCxxFlags = function (cfg)
+				local flags = gmake.toolsets.compilers.gcc.getCFlags(cfg)
+				-- TODO: Add C++ specific flags
+				return flags
+			end
+		}
+	},
+	linkers = {
+		gcc = {
+			getLinkerFlags = function(cfg)
+				local flags = {}
+				if cfg.gmake_architecture == 'x86_64' then
+					table.insert(flags, '-m64')
+					-- should only link on linux
+					table.insert(flags, '-L/usr/lib64')
+				elseif cfg.gmake_architecture == 'x86' then
+					table.insert(flags, '-m32')
+					-- should only link on linux
+					table.insert(flags, '-L/usr/lib32')
+				end
+				return flags
+			end
+		}
+	}
+}
+
+
+---
+-- Exports the GNU makefile for all workspaces.
+---
+function gmake.export()
+	printf('Configuring...')
+	local root = gmake.buildDom()
+
+	for _, wks in ipairs(root.workspaces) do
+		printf('Exporting %s...', wks.name)
+		gmake.exportWorkspace(wks)
+	end
+	print('Done.')
+end
+
+---
+-- Query and build a DOM hierarchy from the contents of the user project script.
+--
+-- @returns
+--	A `dom.Root` object, extended with a few Xcode specific values.
+---
+function gmake.buildDom()
+	local root = dom.Root.new({
+		action = 'gmake'
+	})
+
+	root.workspaces = root:fetchWorkspaces(gmake.fetchWorkspace)
+	return root
+end
+
+---
+-- Fetch the settings for a specific workspace by name, adding values required by
+-- the Visual Studio exporter methods. Also fetches the projects and configurations
+-- used by the workspace.
+--
+-- @param root
+--	A `dom.Root` representing the current root state.
+-- @param name
+--	The name of the workspace to fetch.
+-- @returns
+--	A `dom.Workspace`, with additional Visual Studio specific values.
+---
+function gmake.fetchWorkspace(root, name)
+	local wks = dom.Workspace.new(root
+		:select({ workspaces = name })
+		:withInheritance()
+	)
+
+	wks.root = root
+
+	wks.exportPath = gmake.wks.filename(wks)
+
+	wks.configs = wks:fetchConfigs(gmake.fetchWorkspaceConfig)
+	wks.projects = wks:fetchProjects(gmake.fetchProject)
+
+	table.sort(wks.configs, function(cfg0, cfg1)
+		return (cfg0.gmake_identifier:lower() < cfg1.gmake_identifier:lower())
+	end)
+
+	return wks
+end
+
+
+---
+-- Fetch the settings for a specific project by name, adding values required by
+-- the Visual Studio exporter methods. Also fetches the configurations used by
+-- the project.
+--
+-- @param wks
+--	The `dom.Workspace` instance which contains the target project.
+-- @param name
+--	The name of the project to fetch.
+-- @returns
+--	A `dom.Project`, with additional Visual Studio specific values.
+---
+function gmake.fetchProject(wks, name)
+	local prj = dom.Project.new(wks
+		:select({ projects = name })
+		:fromScopes(wks.root)
+		:withInheritance()
+	)
+
+	prj.root = wks.root
+	prj.workspace = wks
+
+	local name = os.matchFiles(path.join(prj.location, 'Makefile')) and prj.name .. '.mak' or 'Makefile'
```

Comment:
Would be more efficient to use `os.isFile()` here instead of `os.matchFiles()` (I know I still need to document this stuff).

Reply:
Was able to remove this entirely.